controllers/reaper/reaper_controller.go

```diff
@@ -271,63 +273,95 @@ func (r *ReaperReconciler) reconcileService(
 
 func (r *ReaperReconciler) configureReaper(ctx context.Context, actualReaper *reaperapi.Reaper, actualDc *cassdcapi.CassandraDatacenter, logger logr.Logger) (ctrl.Result, error) {
 	manager := r.NewManager()
-	if err := manager.Connect(actualReaper); err != nil {
-		logger.Error(err, "failed to connect to reaper instance")
-		return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
-	} else if found, err := manager.VerifyClusterIsConfigured(ctx, actualDc); err != nil {
-		logger.Error(err, "failed to verify the cluster is registered with reaper")
+	// Get the Reaper UI secret username and password values if auth is enabled
+	if username, password, err := r.getReaperUISecret(ctx, actualReaper, logger); err != nil {
 		return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
-	} else if !found {
-		logger.Info("registering cluster with reaper")
-		if err = manager.AddClusterToReaper(ctx, actualDc); err != nil {
-			logger.Error(err, "failed to register cluster with reaper")
-			return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
+	} else {
+		if err := manager.Connect(ctx, actualReaper, username, password); err != nil {
+			logger.Info("Reaper doesn't seem to be running yet")
+			return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+		} else if found, err := manager.VerifyClusterIsConfigured(ctx, actualDc); err != nil {
+			logger.Info("failed to verify the cluster is registered with reaper. Maybe reaper is still starting up.")
+			return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+		} else if !found {
+			logger.Info("registering cluster with reaper")
+			if err = manager.AddClusterToReaper(ctx, actualDc); err != nil {
+				logger.Error(err, "failed to register cluster with reaper")
+				return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
+			}
 		}
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r *ReaperReconciler) getReaperUISecret(ctx context.Context, actualReaper *reaperapi.Reaper, logger logr.Logger) (string, string, error) {
```

Comment:
Nit: `getReaperUICredentials` would be a more accurate name.

Reply:
sounds good, I'll change it.